web/packages/teleterm/src/ui/DocumentCluster/UnifiedResources.tsx

```diff
@@ -455,6 +463,27 @@ const mapToSharedResource = (
     case 'app': {
       const { resource: app } = resource;
 
+      return {
+        resource: {
+          kind: 'app' as const,
+          labels: app.labels,
+          name: app.name,
+          id: app.name,
+          addrWithProtocol: getAppAddrWithProtocol(app),
+          awsConsole: app.awsConsole,
+          description: app.desc,
+          friendlyName: app.friendlyName,
+          samlApp: app.samlApp,
+          requiresRequest: resource.requiresRequest,
+        },
+        ui: {
+          ActionButton: <ConnectAppActionButton app={app} />,
+        },
+      };
+    }
+    case 'saml_idp_service_provider': {
```

Comment:
Just to clarify: this is not a real resource that we can get from backend, right? We only had to add this  because `UnifiedResourceResponse` type has been extended?

If I understand the problem correctly, the only we need to do, is to send kind `saml_idp_service_provider` instead of `app` when `app.samlApp` is true. This means that we could pass the "correct" kind only for the access request component, the rest of the app doesn't need to know about it. 
Fortunately, it seems that my recent refactor of `accessRequestService` will play nicely with that ☺️. Please check this branch https://github.com/gravitational/teleport/tree/gzdunek/saml-resource-access-request-refactor.

Reply:
Yes correct. I put similar comment in `ResourcesService` but looks like I should have put that here as well. 

> is to send kind saml_idp_service_provider instead of app when app.samlApp is true

Hmm, I did not fully understand from where to "send" that kind, but looking at your branch now.

Reply:
Oh definitely YES. Thank for the reference! - This is now updated in https://github.com/gravitational/teleport/pull/45186/commits/afd6ba696c1905f8d07bc307d17381e2aa113416

updating `extractResourceRequestProperties` and `getRequiredProperties` was the first thing I tried, but I hit with some type error that I could only resolve by updating `UnifiedResourceResponse` type, which resulted in my first implementation. All I had to do was update `ResourceRequest` type to include `samlApp` field but I surely did miss it. Thank you again!